### DISCUSSION TITLE
matcher.add() type error

### DISCUSSION THREAD

**Discussion by the1gofer** [explosion/spaCy#6906] [Help: Coding & Implementations] [answered]
It seems to be failing on the matcher.add line, and from what I can tell through your documentation, this should be fine.

## How to reproduce the behaviour
```
        matcher = Matcher(Parser.nlp.vocab)
        Matcher.add("CityState", pattern)
        matches = matcher(self.doc)
```
TypeError: descriptor 'add' requires a 'spacy.matcher.matcher.Matcher' object but received a 'str'

## Your Environment
- **spaCy version:** 3.0.1
- **Platform:** Darwin-19.6.0-x86_64-i386-64bit
- **Python version:** 3.7.6
- **Pipelines:** en_core_web_trf (3.0.0)* 
- Operating System: mac o 10.15.7

**Comment by adrianeboyd** [comment, marked as answer]
Hi, you want `matcher.add` instead of trying to call this as a class method. This is a good type of question for our new discussions board, so I'll transfer it there in a minute. You'll get an automatic message that it's been locked, but everything will be copied over and the discussion can continue in the thread there.

I can't find an example with this mistake in the docs, but if you do find a non-working example or other mistake, it's fine to submit that as a bug report in the issue tracker, which we're trying to keep focused on bug reports.

**Reply by adrianeboyd** [reply]
Ah, I forgot to add, for v3 you want to provide a list of patterns instead of a single pattern:

```python
matcher.add(match_id, patterns)
```

See: https://spacy.io/api/matcher

**Reply by the1gofer** [reply]
well here is the entire code in question.  I thought I was passing list to it. 
```
        pattern = [{"ENT_TYPE": "GPE"},{"IS_PUNCT": True},{"ENT_TYPE": "GPE"}]

        matcher = Matcher(Parser.nlp.vocab)
        matcher.add("CityState", pattern)
        matches = matcher(self.doc)
        for match_id in matches:
            print(match_id)
```

**Reply by the1gofer** [reply]
I didn't realize it wasn't sending the list as a list, so I think updating to [pattern] fixed the issue. 

Thanks for the help!
